client/blocks/importer/wordpress/upgrade-plan/upgrade-plan-details.tsx

```diff
@@ -1,24 +1,220 @@
 import { recordTracksEvent } from '@automattic/calypso-analytics';
+import config from '@automattic/calypso-config';
 import {
+	calculateMonthlyPriceForPlan,
 	getPlan,
+	Plan,
 	PLAN_BUSINESS,
 	PLAN_BUSINESS_MONTHLY,
 	isMonthly,
 } from '@automattic/calypso-products';
 import { CloudLogo, Button, PlanPrice } from '@automattic/components';
+import { SitePlanPricing } from '@automattic/data-stores';
+import { useSitePlans } from '@automattic/data-stores/src/plans';
 import { Title } from '@automattic/onboarding';
 import { Plans2023Tooltip, useManageTooltipToggle } from '@automattic/plans-grid-next';
+import styled from '@emotion/styled';
 import { useI18n } from '@wordpress/react-i18n';
 import clsx from 'clsx';
+import { useTranslate } from 'i18n-calypso';
 import React, { useState, useEffect } from 'react';
 import ButtonGroup from 'calypso/components/button-group';
 import QuerySitePlans from 'calypso/components/data/query-site-plans';
 import { useSelectedPlanUpgradeMutation } from 'calypso/data/import-flow/use-selected-plan-upgrade';
+import { SitePlanData } from 'calypso/my-sites/checkout/src/hooks/product-variants';
 import { useSelector } from 'calypso/state';
-import { getSitePlan, getSitePlanRawPrice } from 'calypso/state/sites/plans/selectors';
+import { getSitePlan } from 'calypso/state/sites/plans/selectors';
 import { UpgradePlanFeatureList } from './upgrade-plan-feature-list';
 import { UpgradePlanHostingDetails } from './upgrade-plan-hosting-details';
 
+interface SwitcherProps {
+	selectedPlan: string;
+	onMonthlyPlanClick: () => void;
+	onAnnualPlanClick: () => void;
+}
+
+const UpgradePlanPeriodSwitcher = ( props: SwitcherProps ) => {
+	const { __ } = useI18n();
+
+	if ( config.isEnabled( 'migration-flow/introductory-offer' ) ) {
+		return null;
+	}
+
+	const { selectedPlan, onMonthlyPlanClick, onAnnualPlanClick } = props;
+
+	return (
+		<div className="import__upgrade-plan-period-switcher">
+			<ButtonGroup>
+				<Button
+					borderless
+					className={ clsx( { selected: selectedPlan === PLAN_BUSINESS_MONTHLY } ) }
+					onClick={ onMonthlyPlanClick }
+				>
+					{ __( 'Pay monthly' ) }
+				</Button>
+				<Button
+					borderless
+					className={ clsx( { selected: selectedPlan === PLAN_BUSINESS } ) }
+					onClick={ onAnnualPlanClick }
+				>
+					{ __( 'Pay annually' ) }
+				</Button>
+			</ButtonGroup>
+		</div>
+	);
+};
+
+const Badge = styled.div< { isForIntroOffer?: boolean; isHidden?: boolean } >`
+	text-align: center;
+	white-space: nowrap;
+	font-size: 0.75rem;
+	line-height: 1.25rem;
+	border-radius: 4px;
+	height: 21px;
+	display: inline-block;
+	width: fit-content;
+	letter-spacing: 0.2px;
+	font-weight: 500;
+	text-align: center;
+	padding: 0 12px;
+	background-color: #e6d1ff;
+	color: #350e63;
+	text-transform: none;
+`;
```

Comment:
It may be worth looking into whether it would make more sense to re-use an existing badge/pill component for this, rather than generating a completely new entity that will drift from the standard/expected behaviour over time.
How do things look if we use the `Badge` component from `@automattic/components` with the `type` attribute set to `info-purple` or `info-blue`?